src/sentry/search/snuba/backend.py

```diff
@@ -402,27 +396,29 @@ def __run_joined_query_and_log_metric(
             connection.close()
 
     try:
-        _side_query_pool.submit(
-            __run_joined_query_and_log_metric,
-            events_only_search_results,
-            builder,
-            projects,
-            retention_window_start,
-            group_queryset,
-            environments,
-            sort_by,
-            limit,
-            cursor,
-            count_hits,
-            paginator_options,
-            search_filters,
-            date_from,
-            date_to,
-            max_hits,
-            referrer,
-            actor,
-            aggregate_kwargs,
-        )
+        side_query_pool = ThreadPoolExecutor(thread_name_prefix=__name__, max_workers=10)
```

Comment:
Are you sure this doesn't cause us more expensive queries here? Seems odd to make a new thread pool for every query